core/app/models/spree/order.rb

```diff
@@ -692,6 +693,18 @@ def after_cancel
       payments.completed.each { |payment| payment.cancel! }
       send_cancel_email
       self.update!
+
+      # Free up authorized store credits
+      payments.store_credits.pending.each { |payment| payment.void! }
+
+      # payment_state has to be updated because after_cancel on
```

Comment:
this seems odd to introduce in the context of store credits. they ought to be behaving the same as credit card payments, and anything to the contrary should be handled in the payment method, not here imho